src/test/java/de/tum/in/www1/artemis/LectureIntegrationTest.java

```diff
@@ -0,0 +1,118 @@
+
+    }
+
+    @AfterEach
+    public void resetDatabase() {
+        database.resetDatabase();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void getLectureWithPermission() throws Exception {
+        Lecture lecture = database.createCourseWithLecture();
+        lectureRepo.save(lecture);
+        Lecture savedLecture = lectureRepo.findAll().get(0);
+
+        Lecture receivedLecture = request.get("/api/lectures/" + lecture.getId(), HttpStatus.OK, Lecture.class);
+
+        assertThat(savedLecture.getId()).isEqualTo(receivedLecture.getId());
+        assertThat(savedLecture).isEqualTo(receivedLecture);
+
+    }
+
+    @Test
+    @WithMockUser(roles = "USER")
+    public void getLectureForCourseNoPermission() throws Exception {
+        Lecture lecture = database.createCourseWithLecture();
+        lectureRepo.save(lecture);
+        Course course = lectureRepo.findAll().get(0).getCourse();
+        request.get("/api/courses/" + course.getId() + "/lectures", HttpStatus.FORBIDDEN, Lecture.class);
+    }
+
+    @Test
+    @WithMockUser(roles = "USER")
+    public void createLectureNoPermission() throws Exception {
+        Lecture lecture = database.createCourseWithLecture();
+        request.postWithResponseBody("/api/lectures", lecture, Lecture.class, HttpStatus.FORBIDDEN);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void createLectureWithPermission() throws Exception {
+        Lecture lecture = database.createCourseWithLecture();
+        Lecture receivedLecture = request.postWithResponseBody("/api/lectures", lecture, Lecture.class);
+
+        assertThat(receivedLecture).isNotNull();
+        assertThat(receivedLecture.getId()).isNotNull();
+        assertThat(receivedLecture.getDescription()).isEqualTo("Test Lecture");
+
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void updateLectureWithPermission() throws Exception {
+        Lecture lecture = database.createCourseWithLecture();
+        lectureRepo.save(lecture);
+        request.putWithResponseBody("/api/lectures", lecture, Lecture.class, HttpStatus.OK);
+    }
+
+    @Test
+    @WithMockUser(roles = "USER")
+    public void updateLectureNoPermission() throws Exception {
+        Lecture lecture = database.createCourseWithLecture();
+        lectureRepo.save(lecture);
+        request.putWithResponseBody("/api/lectures", lecture, Lecture.class, HttpStatus.FORBIDDEN);
+
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void deleteLectureWithPermission() throws Exception {
+        Lecture lecture = database.createCourseWithLecture();
+        lectureRepo.save(lecture);
+        request.delete("/api/lectures/" + lecture.getId(), HttpStatus.OK);
+        assertThat(lectureRepo.findAll().isEmpty()).isTrue();
+
+    }
+
+    @Test
+    @WithMockUser(roles = "USER")
```

Comment:
Can tutors delete a lecture? If not, maybe add another test or change the role.

Reply:
Yes, they do not have permission to delete a lecture. However, since `TA` is a subset of `USER` I would argue that it should be covered by this test case. I could still add a separate test tho as it has no bad effect :)

Reply:
I don't think it's covered: It could both be true that the test succeeds and that a tutor is able to delete a lecture since a regular user might be denied but a tutor not.

Reply:
I don't think tutors should be able to delete lectures. If this is possible, we should change it so that only instructors can delete it.